src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -408,31 +404,4 @@ export class VSCodeNotebookController implements Disposable {
             await newKernel.start({ disableUI: true }).catch(noop);
         }
     }
-    /**
-     * In our tests, preferred controllers are setup as the active controller.
-     *
-     * This method is called on when running tests, else in the real world,
-     * users need to select a kernel (preferred is on top of the list).
-     */
-    private async setAsActiveControllerForTests(notebook: NotebookDocument) {
```

Comment:
Not sure why thi was deleted, was this to do with getting tets to pass?

Reply:
Because it doesn't work (notebook.selectKernel has to be called in a loop) and it masks other problems (it always associates the controller with the notebook even when it doesn't work).

I moved all of this same logic into the test helper.ts which does select in a loop.

Reply:
The new way seems more likely to find issues as it forces the UI selection change before the controller actually changes.

Reply:
Dont' think tests will pass without this, as without this users will be prompted to select a kernel (this ensures the kernel picker isnt displayed in tests.)

Reply:
No it does the same thing it used to do. It sends the command 'notebook.selectKernel' in a loop. This always happens in all tests before runnign a cell. The old code used to call this setAsActiveControllerForTests instead. Now it just waits for the UI to work.

Reply:
Meaning the selection to change (not for the select kernel picker to come up). 'notebook.selectKernel' does not put up the picker if you pass an id.

Reply:
Yup, found it. All good